server/cloudformation/apps.go

```diff
@@ -0,0 +1,88 @@
+package cloudformation
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/jinzhu/gorm"
+	"github.com/remind101/empire"
+)
+
+// appClient mocks the Empire interface we use.
+type appClient interface {
+	AppsFind(empire.AppsQuery) (*empire.App, error)
+	Create(context.Context, empire.CreateOpts) (*empire.App, error)
+	Destroy(context.Context, empire.DestroyOpts) error
+}
+
+// AppProperties represents the properties for the Custom::EmpireApp
+type AppProperties struct {
+	Name string
+}
+
+// AppResource is a Provisioner that will manage an Empire application
+type AppResource struct {
+	empire appClient
+}
+
+func (p *AppResource) Provision(req Request) (id string, data interface{}, err error) {
+	ctx := context.Background()
+	user := NewUser()
+
+	properties := req.ResourceProperties.(*AppProperties)
```

Comment:
This should never fail, unless it's a programmer error, so I think it's fine to just let it panic in that case.